lntest/itest/lnd_network_test.go

```diff
@@ -278,3 +280,46 @@ func connect(ctxt context.Context, node *lntest.HarnessNode,
 	}
 	return nil
 }
+
+// testAddPeerConfig tests that the "--addpeer" config flag successfully adds
+// a new peer.
+func testAddPeerConfig(net *lntest.NetworkHarness, t *harnessTest) {
+	ctxb := context.Background()
+
+	ctxt, cancel := context.WithTimeout(ctxb, defaultTimeout)
+	defer cancel()
+	alice := net.Alice
+	info, err := alice.GetInfo(ctxt, &lnrpc.GetInfoRequest{})
+	require.NoError(t.t, err)
+
+	alicePeerAddress := info.Uris[0]
+
+	// Create a new node (Carol) with Alice as a peer.
+	args := []string{
+		fmt.Sprintf("--addpeer=%v", alicePeerAddress),
+	}
+	carol := net.NewNode(t.t, "Carol", args)
+	defer shutdownAndAssert(net, t, carol)
+
+	time.Sleep(time.Second)
```

Comment:
You could also just use `assertConnected` here that will wait for a bit, so we don't need to manually sleep.